src/main/java/io/supertokens/session/Session.java

```diff
@@ -171,17 +177,18 @@ public static SessionInformationHolder regenerateToken(AppIdentifier appIdentifi
 
         // We assume the token has already been verified at this point. It may be expired or JWT signing key may have
         // changed for it...
-        AccessTokenInfo accessToken = AccessToken.getInfoFromAccessTokenWithoutVerifying(appIdentifier, token);
-        TenantIdentifier tenantIdentifier = accessToken.tenantIdentifier;
-        if (!tenantIdentifier.toAppIdentifier().equals(appIdentifier)) {
+        AccessTokenInfo accessToken = AccessToken.getInfoFromAccessTokenWithoutVerifying(appIdentifier, main, token);
+        TenantIdentifierWithStorage tenantIdentifierWithStorage = accessToken.tenantIdentifier.withStorage(
+                StorageLayer.getStorage(accessToken.tenantIdentifier, main));
+        if (!tenantIdentifierWithStorage.toAppIdentifier().equals(appIdentifier)) {
             throw new UnauthorisedException("Access token is from an incorrect app");
         }
```

Comment:
this check goes inside getInfoFromAccessTokenWithoutVerifying

Reply:
Done